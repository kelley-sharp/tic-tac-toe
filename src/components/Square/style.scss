.square {
  height: 100px;
  width: 100px;
  padding: 0.5em;
  border: 3px solid;
  border-color: gray;
  display: flex;
  justify-content: center;
  align-items: center;
}

.mark {
  font-size: 64px;
}
