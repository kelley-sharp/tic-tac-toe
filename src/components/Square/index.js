import React, { PureComponent } from 'react';
import './style.scss';

class Square extends PureComponent {
  render() {
    const squareClass = this.props.value ? `square` : `empty-square square`;
    return (
      <div className={squareClass} onClick={this.props.handleClick}>
        <span className="mark">{this.props.value}</span>
      </div>
    );
  }
}

export default Square;
