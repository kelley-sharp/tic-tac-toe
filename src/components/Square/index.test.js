import React from 'react';
import Square from '.';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<Square />', () => {
  // smoke test
  it('renders without crashing', function() {
    shallow(<Square />);
  });

  // snapshot test
  it('matches snapshot', function() {
    let wrapper = shallow(<Square />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});
