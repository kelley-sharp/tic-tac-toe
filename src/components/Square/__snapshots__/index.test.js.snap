// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Square /> matches snapshot 1`] = `
<div
  className="empty-square square"
>
  <span
    className="mark"
  />
</div>
`;
